sample_project/Source/sample_project/Feature Layer/FeatureLayer.cpp

```diff
@@ -27,42 +27,113 @@ AFeatureLayer::AFeatureLayer()
 
 void AFeatureLayer::OnResponseRecieved(FHttpRequestPtr Request, FHttpResponsePtr Response, bool bConnectedSucessfully)
 {
-	TSharedPtr<FJsonObject> ResponseObj;
-	const FString ResponseBody = Response->GetContentAsString();
-	TSharedRef<TJsonReader<>> Reader = TJsonReaderFactory<>::Create(ResponseBody);
-	if (FJsonSerializer::Deserialize(Reader, ResponseObj))
+	if(bConnectedSucessfully)
 	{
-		TArray<TSharedPtr<FJsonValue>> Features = ResponseObj->GetArrayField("features");
-		for (int i = 0; i != Features.Num(); i++)
+		bLinkReturnError = false;
+		TSharedPtr<FJsonObject> ResponseObj;
+		const auto ResponseBody = Response->GetContentAsString();
+		auto Reader = TJsonReaderFactory<>::Create(ResponseBody);
+	
+		if (FJsonSerializer::Deserialize(Reader, ResponseObj))
 		{
-			TArray<double> geoCoordinates = {};
-			TSharedPtr<FJsonObject> feature = Features[i]->AsObject();
-			TSharedPtr<FJsonObject> properties = feature->GetObjectField("properties");
-			TSharedPtr<FJsonObject> Geometry = feature->GetObjectField("geometry");
-			TArray<TSharedPtr<FJsonValue>> coordinates = Geometry->GetArrayField("coordinates");
-			data.NAME.Add(properties->GetStringField("NAME"));
-			data.LEAGUE.Add(properties->GetStringField("LEAGUE"));
-			data.TEAM.Add(properties->GetStringField("TEAM"));
-			data.longitude.Add(coordinates[0]->AsNumber());
-			data.latitude.Add(coordinates[1]->AsNumber());
+			TArray<TSharedPtr<FJsonValue>> Features = ResponseObj->GetArrayField("features");
+
+			for (int i = 0; i != Features.Num(); i++)
+			{
+				FFeatureLayerProperties testProperties;
+				auto feature = Features[i]->AsObject();
+				auto properties = feature->GetObjectField("properties");
+			
+				for (auto outfield : WebLink.OutFields)
+				{
+					testProperties.FeatureProperties.Add(feature->GetObjectField("properties")->GetStringField(outfield));
+				}
+				auto type = feature->GetObjectField("geometry")->GetStringField("type");
+				TArray<TSharedPtr<FJsonValue>> coordinates = feature->GetObjectField("geometry")->GetArrayField("coordinates");
+
+				if(type.ToLower() == "point")
+				{
+					for (const auto Coordinate : coordinates)
+					{
+						testProperties.GeoProperties.Add(Coordinate->AsNumber());
+					}
+					bCoordinatesErrorReturn = false;
+				}
+				else
+				{
+					bCoordinatesErrorReturn = true;
+				}
+				FeatureData.Add(testProperties);
+			}
+		}	
+	}
+	else
+	{
+		bLinkReturnError = true;
+	}
+}
+
+bool AFeatureLayer::ErrorCheck()
+{
+	if(FeatureData.IsEmpty())
+	{
+		bLinkReturnError = true;
+		return false;
+	}
+	
+	for (const auto Data : FeatureData)
+	{
+		FFeatureLayerProperties feature = Data;
+		
+		for (const auto property : feature.FeatureProperties)
```

Comment:
You shouldn't have to need to create a separate local var

```suggestion
	for (const auto feature : FeatureData)
	{		
		for (const auto property : feature.FeatureProperties)
```